python-checks/src/test/resources/checks/verifiedSslTlsCertificateCheck.py

```diff
@@ -0,0 +1,282 @@
+  #                                               ^^^^^
+  requests.options('https://example.domain', verify=False) # Noncompliant {{Disabling certificate verification is dangerous.}}
+  #                                                 ^^^^^
+
+  requests.request(method='GET', url='https://example.domain') # Compliant
+  requests.request(method='GET', url='https://example.domain', verify=True) # Compliant
+  requests.request('GET', 'https://example.domain', verify='/path/to/CAbundle') # Compliant
+  requests.request(verify=True, method='GET', url='https://example.domain') # Compliant
+  kargs = {'verify': True}
+  requests.request('GET', 'https://example.domain', **kargs) # Compliant
+  kargs = {'method': 'GET', 'url': 'https://example.domain', 'verify': True}
+  requests.request(**kargs) # Compliant
+
+  requests.head(url='https://example.domain') # Compliant
+  requests.get(url='https://example.domain') # Compliant
+  requests.post(url='https://example.domain') # Compliant
+  requests.put(url='https://example.domain') # Compliant
+  requests.patch(url='https://example.domain') # Compliant
+  requests.delete(url='https://example.domain') # Compliant
+  requests.options(url='https://example.domain') # Compliant
+  requests.request('GET', 'https://example.domain', verify=range(42)) # Compliant
+  requests.request('GET', 'https://example.domain', verify=range(2, 5)) # Compliant
+
+  # Pathological cases for test coverage
+  requests.request('GET', 'https://example.domain', verify=thatThingIsNotACollectionConstructor(0))
+  requests.request('GET', 'https://example.domain', verify=sorted([])) # FN, bool(sorted([])) == False
+  ft = { 'from': 10, 'to': 100 }
+  requests.request('GET', 'https://example.domain', verify=range(**ft))
+  requests.request('GET', 'https://example.domain', verify=range("not numeric"))
+  requests.request('GET', 'https://example.domain', verify=set(42))
+
+
+def urllibTests():
+  # Mutably borrowed from
+  # https://raw.githubusercontent.com/SonarSource/security-expected-issues/master/python/rules/vulnerabilities/\
+  # RSPEC-4830%20Server%20certificates%20should%20be%20verified%20during%20SSL%E2%81%84TLS%20connections/\
+  # urllib-test.py
+  import urllib.request
+  import ssl
+  import sys
+
+  try:
+    print("test1-default")
+    # (S4830) - bydefault = ctx.verify_mode = ssl.CERT_NONE
+    ctx1 = ssl._create_unverified_context()  # Noncompliant {{Certificate verification is disabled by default, verify_mode should be updated.}}
+    #          ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    r1 = urllib.request.urlopen('https://self-signed.badssl.com/', context=ctx1)
+    print(r1.status, r1.reason)
+
+    print("test1-1")
+    ctx2 = ssl._create_unverified_context()
+    ctx2.verify_mode = ssl.CERT_NONE # Noncompliant {{Disabling certificate verification is dangerous.}}
+    #                      ^^^^^^^^^
+    r1 = urllib.request.urlopen('https://self-signed.badssl.com/', context=ctx2)
```

Comment:
Same here, I would preferer keeping the primary location to `ctx2` and adding a secondary location

Reply:
I went through the test python file again, and made sure that all the issue positions are exactly on the same lines as in the example attached to the specification:
https://github.com/SonarSource/security-expected-issues/blob/master/python/rules/vulnerabilities/RSPEC-4830%20Server%20certificates%20should%20be%20verified%20during%20SSL%E2%81%84TLS%20connections/urllib-test.py
In the specification, it seems that the last position where `verify_mode` is set to `CERT_NONE` is marked as noncompliant. In case of factory method invocations with bad defaults and without subsequent overriding of `verify_mode`, this simply reduces to underlining the factory method itself. This seems consistent.

Reply:
Makes sense. Thanks!